### DISCUSSION TITLE
Validation schema

### DISCUSSION THREAD

**Discussion by weber8thomas** [GMOD/jbrowse-components#4202] [Q&A]
Hi, 

I wanted to know if there were existing schemas that could help validated a template JSON at the track level. I'd allow users to leverage jbrowse configurability through templates like this below and wouold like to make sure the configuration provided respect or not JBrowse requirements 

```
{
    "type": "FeatureTrack",
    "name": "{cell}-SV",
    "trackId": "{trackId}",
    "assemblyNames": [
      "hg38"
    ],
    "category": [
        "SV calls"
      ],
    "adapter": {
      "type": "BedTabixAdapter",
      "bedGzLocation": {
        "locationType": "UriLocation",
        "uri": "{uri}"
      },
      "index": {
        "location": {
          "locationType": "UriLocation",
          "uri": "{indexUri}"
        }
      }
    },
    "displays": [
      {
        "type": "LinearBasicDisplay",
        "displayId": "{displayId}",
        "height": 20,
        "renderer": {
          "type": "SvgFeatureRenderer",
          "color1": "jexl:get(feature,\"field4\")",
          "height": 20
        }
      }
    ]
  }

```

Many thanks in advance!

Thomas

**Comment by cmdcolin** [comment]
this is a great idea. I have considered that maybe having something like a json-schema to validate would be good. it is tough because the actual typescript is not very equipped to check these schemas, but, we have our autogenerated docs that are generated from the typescript types (e.g. https://jbrowse.org/jb2/docs/config/deltaadapter/) and i feel like these could generate json schema potentially also, though i haven't investigated much
